lib/vector-core/src/event/vrl_target.rs

```diff
@@ -20,18 +21,18 @@ const MAX_METRIC_PATH_DEPTH: usize = 3;
 pub enum VrlTarget {
     // `LogEvent` is essentially just a destructured `event::LogEvent`, but without the semantics
     // that `fields` must always be a `Map` variant.
-    LogEvent(Value, EventMetadata),
+    LogEvent(SharedValue, EventMetadata),
     Metric(Metric),
 }
 
 impl VrlTarget {
-    pub fn new(event: Event) -> Self {
+    pub fn new(event: &Event) -> Self {
         match event {
-            Event::Log(event) => {
-                let (fields, metadata) = event.into_parts();
-                VrlTarget::LogEvent(Value::Map(fields), metadata)
-            }
-            Event::Metric(event) => VrlTarget::Metric(event),
+            Event::Log(event) => VrlTarget::LogEvent(
+                SharedValue::from(vrl_core::Value::from(&event.fields)),
```

Comment:
`VrlTarget` is now a `SharedValue` (a wrapper over `Rc<RefCell<Value>>`). When creating the target we need to copy the event into this `SharedValue`. This is why the clone is no longer necessary since the original event is then untouched until the end of the process.